pkg/daemon/update_test.go

```diff
@@ -353,22 +386,84 @@ func TestReconcilableSSH(t *testing.T) {
 	checkIrreconcilableResults(t, "SSH", errMsg)
 }
 
-func TestUpdateSSHKeys(t *testing.T) {
-	// testClient is the NodeUpdaterClient mock instance that will front
-	// calls to update the host.
-	testClient := RpmOstreeClientMock{
-		GetBootedOSImageURLReturns: []GetBootedOSImageURLReturn{},
+func TestWriteFiles(t *testing.T) {
+	testDir, cleanup := setupTempDirWithEtc(t)
+	defer cleanup()
+
+	d := newMockDaemon()
+
+	contents := []byte("hello world\n")
+	encodedContents := dataurl.EncodeBytes(contents)
+
+	// gzip contents
+	var buffer bytes.Buffer
+	writer := gzip.NewWriter(&buffer)
+	_, err := writer.Write(contents)
+	assert.Nil(t, err)
+	err = writer.Close()
+	assert.Nil(t, err)
+	gzippedContents := dataurl.EncodeBytes(buffer.Bytes())
+
+	// use current user so test doesn't try to chown to root
+	currentUser, err := user.Current()
+	assert.Nil(t, err)
+	currentUid, err := strconv.Atoi(currentUser.Uid)
+	assert.Nil(t, err)
+	currentGid, err := strconv.Atoi(currentUser.Gid)
+	assert.Nil(t, err)
+
+	filePath := testDir + "test"
+	node := ign3types.Node{
+		Path:  filePath,
+		User:  ign3types.NodeUser{ID: &currentUid},
+		Group: ign3types.NodeGroup{ID: &currentGid},
 	}
 
-	// Create a Daemon instance with mocked clients
-	d := Daemon{
-		mock:              true,
-		name:              "nodeName",
-		os:                OperatingSystem{},
-		NodeUpdaterClient: testClient,
-		kubeClient:        k8sfake.NewSimpleClientset(),
-		bootedOSImageURL:  "test",
+	mode := 420
+
+	tests := []struct {
+		files            []ign3types.File
+		expectedErr      error
+		expectedContents []byte
+	}{
+		{
+			files: []ign3types.File{{
+				Node:          node,
+				FileEmbedded1: ign3types.FileEmbedded1{Contents: ign3types.Resource{Source: &encodedContents}, Mode: &mode},
+			}},
+			expectedContents: contents,
+		},
+		{
+			files: []ign3types.File{{
+				Node:          node,
+				FileEmbedded1: ign3types.FileEmbedded1{Contents: ign3types.Resource{Source: &gzippedContents, Compression: helpers.StrToPtr("gzip")}, Mode: &mode},
+			}},
+			expectedContents: contents,
+		},
+		{
+			files: []ign3types.File{{
+				Node:          node,
+				FileEmbedded1: ign3types.FileEmbedded1{Contents: ign3types.Resource{Source: &encodedContents, Compression: helpers.StrToPtr("xz")}, Mode: &mode},
+			}},
+			expectedErr: fmt.Errorf("unsupported compression type %q for file %q", "xz", filePath),
+		},
+	}
+
+	for _, test := range tests {
+		err := d.writeFiles(test.files)
```

Comment:
Once you have the `name` field added, wrap the loop contents in a `t.Run()` function, e.g.:

```golang
for _, test := range tests {
  t.Run(test.name, func(t *testing.T) {
    // Loop contents go here
  })
}
```

The reason for this is because if one (or more) of those testcases fails, it will be easier to figure out which one failed when you look in the test output.